ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/fragment/PaywallFooterView.kt

```diff
@@ -0,0 +1,137 @@
+import com.revenuecat.purchases.ui.revenuecatui.PaywallFooter
+import com.revenuecat.purchases.ui.revenuecatui.PaywallListener
+import com.revenuecat.purchases.ui.revenuecatui.PaywallOptions
+import com.revenuecat.purchases.ui.revenuecatui.R
+import com.revenuecat.purchases.ui.revenuecatui.fonts.CustomFontProvider
+import com.revenuecat.purchases.ui.revenuecatui.fonts.FontProvider
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+
+/**
+ * View that wraps the [PaywallFooter] Composable to display the Paywall Footer through XML layouts and the View system.
+ */
+@ExperimentalPreviewRevenueCatUIPurchasesAPI
+class PaywallFooterView : FrameLayout {
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init(context, attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        init(context, attrs)
+    }
+
+    /**
+     * Constructor when creating the view programmatically.
+     */
+    @JvmOverloads
+    constructor(
+        context: Context,
+        offering: Offering? = null,
+        listener: PaywallListener? = null,
+        fontProvider: FontProvider? = null,
+        dismissHandler: (() -> Unit)? = null,
+    ) : super(context) {
+        setPaywallListener(listener)
+        setDismissHandler(dismissHandler)
+        setOfferingId(offering?.identifier)
+        this.fontProvider = fontProvider
+        init(context, null)
+    }
+
+    private var offeringId: String? = null
+    private var fontProvider: FontProvider? = null
+    private var dismissHandler: (() -> Unit)? = null
+    private var listener: PaywallListener? = null
+    private var internalListener: PaywallListener = object : PaywallListener {
+        override fun onPurchaseStarted(rcPackage: Package) { listener?.onPurchaseStarted(rcPackage) }
+        override fun onPurchaseCompleted(customerInfo: CustomerInfo, storeTransaction: StoreTransaction) {
+            listener?.onPurchaseCompleted(customerInfo, storeTransaction)
+        }
+        override fun onPurchaseError(error: PurchasesError) { listener?.onPurchaseError(error) }
+        override fun onRestoreStarted() { listener?.onRestoreStarted() }
+        override fun onRestoreCompleted(customerInfo: CustomerInfo) { listener?.onRestoreCompleted(customerInfo) }
+        override fun onRestoreError(error: PurchasesError) { listener?.onRestoreError(error) }
+    }
+
+    /**
+     * Sets a [PaywallListener] to receive callbacks from the Paywall.
+     */
+    fun setPaywallListener(listener: PaywallListener?) {
+        this.listener = listener
+    }
+
+    /**
+     * Sets a dismiss handler which will be called when the user successfully purchases or if there is an error
+     * loading the offerings and the user clicks through the error dialog.
+     */
+    fun setDismissHandler(dismissHandler: (() -> Unit)?) {
```

Comment:
I wondered whether to remove this for the footer view, since there is already a purchase completed listener in the `PaywallListener`. But there can still be an error loading the offerings and we show an error dialog in that case, In that case, we want to dismiss it, and this would be the only way to do it.